libs/providers/multi-provider/src/lib/multi-provider.ts

```diff
@@ -0,0 +1,325 @@
+import {
+  DefaultLogger,
+  EvaluationContext,
+  FlagValueType,
+  GeneralError,
+  Hook,
+  HookContext,
+  HookHints,
+  JsonValue,
+  Logger,
+  OpenFeatureEventEmitter,
+  Provider,
+  ResolutionDetails,
+} from '@openfeature/server-sdk';
+import { BeforeHookContext, ProviderMetadata } from '@openfeature/core';
+import { HookExecutor } from './hook-executor';
+import { constructAggregateError } from './errors';
+import { BaseEvaluationStrategy, ProviderResolutionResult, FirstMatchStrategy } from './strategies';
+import { StatusTracker } from './status-tracker';
+
+// Represents an entry in the constructor's provider array which may or may not have a name set
+export type ProviderEntryInput = {
+  provider: Provider;
+  name?: string;
+};
+
+// Represents a processed and "registered" provider entry where a name has been chosen
+export type RegisteredProvider = {
+  name: string;
+  provider: Provider;
+};
+
+export class MultiProvider implements Provider {
+  readonly runsOn = 'server';
+
+  public readonly events = new OpenFeatureEventEmitter();
+
+  private hookContexts: WeakMap<EvaluationContext, HookContext> = new WeakMap<EvaluationContext, HookContext>();
+  private hookHints: WeakMap<EvaluationContext, HookHints> = new WeakMap<EvaluationContext, HookHints>();
+
+  metadata: ProviderMetadata;
+
+  providerEntries: RegisteredProvider[] = [];
+  private providerEntriesByName: Record<string, RegisteredProvider> = {};
```

Comment:
I would prefer `private providerEntriesByName: ReadonlyMap<string, RegisteredProvider> = {};` here.
This way we would also not have to have `providerEntries` maintained, we could just have 

```ts
  public get providerEntries(): RegisteredProvider[] {
    return [...this.providerEntriesByName.values()];
  }
```
which is less error prone, in the case we will later maybe update `providerEntriesByName`.
The only problem is, `ReadonlyMap` only applies to TS and we can not freeze the Map.

Reply:
Ah good point, makes sense!